RoboJackets/robocup-software
# Scrim 2022 merge

## Description
Merges in all useful changes from #1887 . See that PR for [list of changes](https://github.com/RoboJackets/robocup-software/pull/1887#pullrequestreview-959588745) I chose to drop from the virtual scrim.

## Steps to test
Run ER-force sim + ssl-game-controller + `make run-sim-ex` and play through some different situations.

Expected result: Nothing crashes, no excess debug prints are shown, gameplay looks more or less as it did during scrim 2022.
